plugins/synced_folders/rsync/helper.rb

```diff
@@ -213,6 +213,10 @@ def self.rsync_single(machine, ssh_info, opts)
           r = Vagrant::Util::Subprocess.execute(*(command + [command_opts]))
         end
 
+        unless Vagrant::Util::Platform.windows?
+          FileUtils.remove_entry_secure(controlpath)
+        end
```

Comment:
Can we instead move this to the end of the function with an `ensure`? Then we can be sure this directory is removed just in case an exception is thrown before we reach here. You'll also want to make sure `controlpath` exists in the ensure before removing it just in case the function returns before `controlpath` is created.

Reply:
It's probably also a good thing to have `require 'fileutils'` at the top of this file just in case.